src/mon/Monitor.cc

```diff
@@ -2863,6 +2871,15 @@ void Monitor::log_health(
          << p.first << ")";
```

Comment:
How about adding `p.second.detail` here and later where we update health? The update logging is already rate limited `mon_health_log_update_period` . @jdurgin @sunnyku what do you think?

Reply:
@neha-ojha @jdurgin Updated PR accordingly. The `do_health_to_clog()` will now log -
1. cluster summary or health detail every 10 minutes
2. Overall cluster summary will be logged when cluster is HEALTH_OK or there is no change in summary since last known summary
3. If cluster status is WARN/ERR then `ceph health detail` will be logged and summary will be recorded.
4. If cluster status is still WARN/ERR but there is a change in summary then `ceph health detail` will be logged

This will avoid logging `ceph health detail` too frequently and re-logging same health detail over the period of time.